### PR TITLE
feat (contrib/plugins): add plugin download

### PR DESCRIPTION
**Pull request by bnjjj**
Signed-off-by: Benjamin Coenen <benjamin.coenen@corp.ovh.com>

### CODE REVIEW

File: contrib/plugins/plugin-download/download.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	// Create the file
+	file, err := os.Create(filepath)
+	if err != nil {
+		plugin.SendLog(a, "Error to create the file %s : %s", filepath, err)
+		return plugin.Fail
+	}
+	defer file.Close()
+
+	// Download from URL
+	resp, err := http.Get(url)
```
Comment (fsamin) [inline contrib/plugins/plugin-download/download.go]: you should always set http timeout https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779

File: contrib/plugins/plugin-download/download.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	if err != nil {
+		plugin.SendLog(a, "Error to download the file on URL %s : %s", url, err)
+		return plugin.Fail
+	}
+	defer resp.Body.Close()
+
+	// Copy file in the right directory
+	_, err = io.Copy(file, resp.Body)
+	if err != nil {
```
Comment (fsamin) [inline contrib/plugins/plugin-download/download.go]: one line please

File: contrib/plugins/plugin-download/download.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+func (d DownloadPlugin) Name() string        { return "download" }
+func (d DownloadPlugin) Description() string { return "This is a plugin to download file from URL" }
+func (d DownloadPlugin) Author() string      { return "Benjamin COENEN <benjamin.coenen@corp.ovh.com>" }
+
+// Parameters return parameters description
+func (d DownloadPlugin) Parameters() plugin.Parameters {
+	params := plugin.NewParameters()
+	params.Add("url", plugin.StringParameter, "the url of your file", "{{.cds.app.downloadUrl}}")
+	params.Add("filepath", plugin.StringParameter, "the destination of your file to be copied", ".")
+
```
Comment (fsamin) [inline contrib/plugins/plugin-download/download.go]: maybe we should manage headers like basic auth headers...
add a "headers param" which should be formatted like "Header-Key-1=VALUE Header-Key-2=VALUE"

File: contrib/plugins/plugin-download/download.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	resp, err := httpClient.Do(req)
+
+	if err != nil {
+		plugin.SendLog(a, "Error to download the file on URL %s : %s", url, err)
+		return plugin.Fail
+	}
+	defer resp.Body.Close()
+
+	// Copy file in the right directory
+	if _, err = io.Copy(file, resp.Body); err != nil {
```
Comment (sguiheux) [inline contrib/plugins/plugin-download/download.go]: _, err :=

Comment (bnjjj) [inline contrib/plugins/plugin-download/download.go]: Oups, fixed :)

File: contrib/plugins/plugin-download/download.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+func (d DownloadPlugin) Name() string        { return "download" }
+func (d DownloadPlugin) Description() string { return "This is a plugin to download file from URL" }
+func (d DownloadPlugin) Author() string      { return "Benjamin COENEN <benjamin.coenen@corp.ovh.com>" }
+
+// Parameters return parameters description
+func (d DownloadPlugin) Parameters() plugin.Parameters {
+	params := plugin.NewParameters()
+	params.Add("url", plugin.StringParameter, "the url of your file", "{{.cds.app.downloadUrl}}")
+	params.Add("filepath", plugin.StringParameter, "the destination of your file to be copied", ".")
+	params.Add("headers", plugin.StringParameter, `specific headers to add to your request ("headerName"="value" newline separated list)`, "")
```
Comment (yesnault) [inline contrib/plugins/plugin-download/download.go]: TextParameter will let user to allow new line.

Comment (bnjjj) [inline contrib/plugins/plugin-download/download.go]: Done :) Thank you !

**Pull request merged**